src/components/CurriculumComponents/UnitsTabMobile/UnitsTabMobile.tsx

```diff
@@ -0,0 +1,256 @@
+import React, { FC, useState, useRef, useEffect } from "react";
+import { OakP, OakHeading, OakFlex } from "@oaknational/oak-components";
+
+import { Thread } from "../CurriculumVisualiser/CurriculumVisualiser";
+
+import Box from "@/components/SharedComponents/Box";
+import Button from "@/components/SharedComponents/Button/Button";
+import ButtonGroup from "@/components/SharedComponents/ButtonGroup";
+import Radio from "@/components/SharedComponents/RadioButtons/Radio";
+import RadioGroup from "@/components/SharedComponents/RadioButtons/RadioGroup";
+
+// Types and interfaces
+
+type UnitsTabMobileProps = {
+  selectedThread: Thread | null;
+  handleSelectThread: (thread: string) => void;
+  threadOptions: Thread[];
+  isSelectedThread: (thread: Thread) => boolean;
+  highlightedUnitCount: () => number;
+  trackSelectYear: (year: string) => void;
+  yearOptions: string[];
+  updateMobileHeaderScroll: (height: number) => void;
+  visibleMobileYearRefID: string | null;
+};
+
+// Function component
+
+const UnitsTabMobile: FC<UnitsTabMobileProps> = ({
+  updateMobileHeaderScroll,
+  selectedThread,
+  handleSelectThread,
+  threadOptions,
+  isSelectedThread,
+  highlightedUnitCount,
+  trackSelectYear,
+  yearOptions,
+  visibleMobileYearRefID,
+}) => {
+  // Initialize constants
+  const [mobileThreadModalOpen, setMobileThreadModalOpen] =
+    useState<boolean>(false);
+  const [mobileYearSelection, setMobileYearSelection] = useState<string | null>(
+    "7",
```

Comment:
Shouldn't need the "7", if it does then it probably should be fixed in a different location or with a different non hardcoded value.